client/packages/purchase_orders/src/DetailView/LineEdit/PurchaseOrderLineEditModal.tsx

```diff
@@ -0,0 +1,126 @@
+import { PurchaseOrderFragment } from '../../api';
+import { DialogButton, InlineSpinner } from '@common/components';
+import { useTranslation, Box } from '@openmsupply-client/common';
+import React, { useState } from 'react';
+import { PurchaseOrderLineEdit } from './PurchaseOrderLineEdit';
+import { usePurchaseOrderLine } from '../../api/hooks/usePurchaseOrderLine';
+import { ItemWithStatsFragment } from '@openmsupply-client/system';
+import { createDraftPurchaseOrderLine } from './utils';
+interface PurchaseOrderLineEditModalProps {
+  itemId: string | null;
+  purchaseOrder: PurchaseOrderFragment;
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const PurchaseOrderLineEditModal = ({
+  itemId,
+  purchaseOrder,
+  mode,
+  isOpen,
+  onClose,
+}: PurchaseOrderLineEditModalProps) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+
+  const lines = purchaseOrder.lines.nodes;
+
+  const [currentItem, setCurrentItem] = useState(
+    lines.find(line => line.item.id === itemId) ?? undefined
+  );
+
+  const {
+    create: { create, isCreating },
+    draft,
+    updatePatch,
+  } = usePurchaseOrderLine(currentItem?.item.id);
+
+  const onChangeItem = (item: ItemWithStatsFragment) => {
+    const draftLine = createDraftPurchaseOrderLine(item, purchaseOrder.id);
+    item &&
+      updatePatch({
+        ...draftLine,
+      });
+    setCurrentItem({
+      ...draftLine,
+      __typename: 'PurchaseOrderLineNode',
+      item: item,
+    });
+  };
+
+  const handleSave = async () => {
+    try {
+      await create();
+      updatePatch(draft);
+      return true;
+      // TODO add proper error handling by returning type errors from API
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        error(e.message)();
+      } else {
+        error('unknown error')();
+      }
+      return false;
+    }
+  };
+
+  // TODO handle next item workflow
+  // const onSave = async () => {
+  //   const success = await handleSave();
+  //   if (!success) return false;
+  //   // if (mode === ModalMode.Update && false setCurrentItem(next);
+  //   // else if (mode === ModalMode.Create) setCurrentItem(undefined);
+  //   // else onClose();
+  //   onClose();
+  //   return true;
+  // };
+
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+
+  return (
+    <Modal
+      title={
+        mode === ModalMode.Create
+          ? t('heading.add-item')
+          : t('heading.edit-item')
+      }
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
+      // TODO add next button functionality
+      // nextButton={<DialogButton variant="next-and-ok" onClick={onSave} />}
+      okButton={
+        <DialogButton
+          variant="ok"
+          disabled={!currentItem}
+          onClick={async () => {
+            const success = await handleSave();
+            if (success) onClose();
```

Comment:
This isn't quite right - our work flow will change to leave the modal open - but in edit mode to change the quantities etc once the line has been inserted.

Reply:
I wonder if it's worth changing the insert line mutation to take all the fields? It's different to insert purchase order when we want to start with a blank one...

Can be follow up PR though.